pkg/asset/agent/manifests/nmstateconfig_test.go

```diff
@@ -0,0 +1,270 @@
+package manifests
+
+import (
+	"errors"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/openshift/assisted-service/models"
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/mock"
+)
+
+func TestNMStateConfig_Generate(t *testing.T) {
+
+	// TODO
+
+}
+
+func TestNMStateConfig_LoadedFromDisk(t *testing.T) {
+
+	cases := []struct {
+		name               string
+		data               string
+		fetchError         error
+		expectedFound      bool
+		expectedError      bool
+		requiresNmstatectl bool
+		expectedConfig     []*models.HostStaticNetworkConfig
+	}{
+		{
+			name: "valid-config-file",
+			data: `
+metadata:
```

Comment:
Not blocking: to improve the test readability, and to make it easier adding new cases, I'd recommend to follow the fluent interface approach for defining the actual scenario as described in [platform_test.go](https://github.com/openshift/installer/blob/3c4a5fb3d859cc150a2dd936517501ba07d5e825/pkg/types/baremetal/validation/platform_test.go#L20)

Reply:
Per our discussion, yes that this is a good idea for readability. As the tests are consistent with the other agent manifests (infraenv_test.go, pullsecret_test.go) I'd like to keep it as is for this PR and do the refratoring later